Provider/src/main/java/com/spotify/confidence/EventSender.kt

```diff
@@ -0,0 +1,40 @@
+package com.spotify.confidence
+
+import com.spotify.confidence.client.serializers.JsonAnySerializer
+import kotlinx.serialization.SerializationStrategy
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import java.io.File
+
+interface EventSender : Contextual {
+    fun send(
+        definition: String,
+        payload: ConfidenceFieldsType = mapOf()
+    )
+
+    fun onLowMemory(body: (List<File>) -> Unit): Contextual
+    fun stop()
```

Comment:
I think it would be best to keep this functionality on the root (parent) Confidence object. Otherwise someone could run this on a child event sender and make all the event sender break, right?

Reply:
I created a `RootConfidence` and `ChildConfidence` and moved the stop functionality to the RootConfidence only like we did in Java. but I don't like this approach so much, I'd rather have each instance signal its own shutdown and the main engine stops what it's doing if all the instances are shut down instead. which I'll tackle it in another PR. is having this `RootConfidence` which its only difference with `Confidence` interface is one additional `shutdown` function good enough for this PR?